apps/prairielearn/elements/pl-integer-input/pl-integer-input.py

```diff
@@ -159,17 +159,25 @@ def render(element_html: str, data: pl.QuestionData) -> str:
             # If answer is a string, convert back to an integer.
             # Else if answer is in a format generated by pl.to_json, convert it
             # back to a standard type. Otherwise do nothing.
-            if isinstance(a_sub, str):
-                a_sub_parsed = int(a_sub)
-            else:
-                a_sub_parsed = pl.from_json(a_sub)
+            try:
+                if isinstance(a_sub, str):
+                    # The representation of the submitted_answer is always
+                    # stored in base 10 by the parse function
+                    a_sub_parsed = int(a_sub)
+                else:
+                    a_sub_parsed = pl.from_json(a_sub)
+                a_sub = (
```

Comment:
If this `a_sub` is only set to be put into the dictionary below, can you use a different variable name, maybe `a_sub_display`? This was initially hard to read because of this.

Reply:
Done in bb1ec39eb64dc7e36b3d38deb7d051691ee6b9c5.